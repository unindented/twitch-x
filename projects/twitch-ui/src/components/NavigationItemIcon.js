import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const NavigationItemIconRoot = styled.span`
  display: block;
  width: ${props => props.theme.layout.navCollapsedWidth};
  height: ${props => props.theme.layout.navCollapsedWidth};
  flex: 0 0 auto;
  padding: ${props => props.theme.layout.gap};
`

const NavigationItemIconImage = styled.img`
  display: block;
  width: 100%;
`

export default function NavigationItemIcon (props) {
  const {icon, iconAlt} = props

  return (
    <NavigationItemIconRoot>
      <NavigationItemIconImage
        src={icon}
        alt={iconAlt}
      />
    </NavigationItemIconRoot>
  )
}

NavigationItemIcon.propTypes = {
  icon: PropTypes.string.isRequired,
  iconAlt: PropTypes.string.isRequired
}
