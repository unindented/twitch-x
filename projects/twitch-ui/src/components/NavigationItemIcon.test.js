import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import NavigationItemIcon from './NavigationItemIcon'

describe('NavigationItemIcon', () => {
  let tree

  beforeEach(() => {
    const renderer = new ShallowRenderer()
    renderer.render(<NavigationItemIcon
      icon='home'
      iconAlt=''
      label='Home'
    />)
    tree = renderer.getRenderOutput()
  })

  it('renders', () => {
    expect(tree).toMatchSnapshot()
  })
})
