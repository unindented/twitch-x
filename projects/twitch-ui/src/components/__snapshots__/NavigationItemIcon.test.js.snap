// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItemIcon renders 1`] = `
<NavigationItemIcon__NavigationItemIconRoot>
  <NavigationItemIcon__NavigationItemIconImage
    alt=""
    src="home"
  />
</NavigationItemIcon__NavigationItemIconRoot>
`;
